lib/zipkin-tracer/zipkin_b3_single_header_format.rb

```diff
@@ -0,0 +1,38 @@
+module ZipkinTracer
+  # This format corresponds to the propagation key "b3" (or "B3").
+  # b3: {x-b3-traceid}-{x-b3-spanid}-{if x-b3-flags 'd' else x-b3-sampled}-{x-b3-parentspanid}
+  # For details, see: https://github.com/apache/incubator-zipkin-b3-propagation
+  class B3SingleHeaderFormat
+    attr_reader :trace_id, :span_id, :parent_span_id, :sampled, :flags
+
+    def initialize(trace_id: nil, span_id: nil, parent_span_id: nil, sampled: nil, flags: Trace::Flags::EMPTY)
+      @trace_id = trace_id
+      @span_id = span_id
+      @parent_span_id = parent_span_id
+      @sampled = sampled
+      @flags = flags
+    end
+
+    def to_a
+      [trace_id, span_id, parent_span_id, sampled, flags]
+    end
+
+    def self.parse_from_header(b3_single_header)
+      return new(parse_sampled_flags(b3_single_header)) if b3_single_header.size == 1
+
+      trace_id, span_id, flag, parent_span_id = b3_single_header.split('-')
+      new(trace_id: trace_id, span_id: span_id, parent_span_id: parent_span_id, **parse_sampled_flags(flag))
```

Comment:
I think it's a bit wasteful to have this method create a new array (by calling split), then a new object and then in the caller call `to_a` which is going to create yet another array.

How about having a method getting the header as an input and return an array directly, without instantiating?

Reply:
though I would say this is the least code approach I've ever seen :P

Reply:
Updated in 2fef91b